sugargame/event.py

```diff
@@ -186,72 +183,76 @@ def _get_mouse_pressed(self):
         # From http://www.learningpython.com/2006/07/25/writing-a-custom-widget-using-pygtk/
-        # if this is a hint, then let's get all the necessary 
+        # if this is a hint, then let's get all the necessary
         # information, if not it's all we need.
         if event.is_hint:
-            x, y, state = event.window.get_pointer()
+            win, x, y, state = event.window.get_device_position(event.device)
         else:
             x = event.x
             y = event.y
-            state = event.state
+            state = event.get_state()
 
         rel = (x - self.__mouse_pos[0], y - self.__mouse_pos[1])
         self.__mouse_pos = (x, y)
-        
+
         self.__button_state = [
-            state & gtk.gdk.BUTTON1_MASK and 1 or 0,
-            state & gtk.gdk.BUTTON2_MASK and 1 or 0,
-            state & gtk.gdk.BUTTON3_MASK and 1 or 0,
+            state & Gdk.ModifierType.BUTTON1_MASK and 1 or 0,
+            state & Gdk.ModifierType.BUTTON2_MASK and 1 or 0,
+            state & Gdk.ModifierType.BUTTON3_MASK and 1 or 0,
         ]
-        
+
         evt = pygame.event.Event(pygame.MOUSEMOTION,
-                                 pos=self.__mouse_pos, rel=rel, buttons=self.__button_state)
+                                 pos=self.__mouse_pos, rel=rel,
+                                 buttons=self.__button_state)
         self._post(evt)
         return True
-        
+
     def _tick_cb(self):
         cur_time = pygame.time.get_ticks()
         for key in self.__held:
-            delta = cur_time - self.__held_last_time[key] 
+            delta = cur_time - self.__held_last_time[key]
             self.__held_last_time[key] = cur_time
-            
+
             self.__held_time_left[key] -= delta
             if self.__held_time_left[key] <= 0:
                 self.__held_time_left[key] = self.__repeat[1]
                 self._keyevent(None, _MockEvent(key), pygame.KEYDOWN)
-                
+
         return True
-        
+
     def _set_repeat(self, delay=None, interval=None):
         if delay is not None and self.__repeat[0] is None:
-            self.__tick_id = gobject.timeout_add(10, self._tick_cb)
+            self.__tick_id = GObject.timeout_add(10, self._tick_cb)
         elif delay is None and self.__repeat[0] is not None:
-            gobject.source_remove(self.__tick_id)
+            GObject.source_remove(self.__tick_id)
         self.__repeat = (delay, interval)
-        
+
     def _get_mouse_pos(self):
         return self.__mouse_pos
 
     def _post(self, evt):
         try:
             pygame.event.post(evt)
         except pygame.error, e:
-            if str(e) == 'Event queue full':
-                print "Event queue full!"
+            if str(e) == 'video system not initialized':
+                pass
+            elif str(e) == 'Event queue full':
+                logging.error("Event queue full!")
                 pass
             else:
                 raise e
```

Comment:
I've not checked your changes to Sugargame v1.2, but it reminds me that there may be activities with a `sugargame/` that has diverged from the original, adding features or fixing bugs, so if and when you find these please contribute them to the Sugargame repository somehow.

Reply:
I will contribute those changes to Sugargame as soon as I find them. 👍